src/providers/Staking.js

```diff
@@ -0,0 +1,36 @@
+import React, { useContext, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import { useStaking } from '../hooks/useStaking'
+
+const StakingContext = React.createContext({
+  staking: {},
+  loading: true,
+})
+
+function StakingProvider({ children }) {
+  const [stakeManagement, loading] = useStaking()
+
+  const Staking = useMemo(
```

Comment:
As per previous comment, let's chat about this as we can probably use the existing `agreementSubscriptionProvider` for this :)